build/downloader.ts

```diff
@@ -108,38 +108,44 @@ async function downloadLanguageServer(platform: string, architecture: string, ex
   });
 }
 
+async function downloadFile(url: string, installPath: string) {
+  if (process.env.downloader_log === 'true') {
+    console.log(`Downloading: ${url}`);
+  }
+
+  const buffer = await fileFromUrl(url);
+  fs.writeFileSync(installPath, buffer);
+  if (process.env.downloader_log === 'true') {
+    console.log(`Download completed: ${installPath}`);
+  }
+}
+
 async function downloadSyntax(info: ExtensionInfo) {
   const release = `v${info.syntaxVersion}`;
 
-  const productName = info.name.replace('-preview', '');
-  const fileName = `${productName}.tmGrammar.json`;
-
   const cwd = path.resolve(__dirname);
   const buildDir = path.basename(cwd);
   const repoDir = cwd.replace(buildDir, '');
   const installPath = path.join(repoDir, 'syntaxes');
 
-  const fpath = path.join(installPath, fileName);
-  if (fs.existsSync(fpath)) {
+  if (fs.existsSync(installPath)) {
```

Comment:
I think if we only check for the existence of the directory, it's easy to run into a situation where one grammar file exists and the other doesn't. `npm run download:artifacts` will then exit silently. Can we improve this behavior?

Reply:
It does have logging if you enable the env variable, but its a valid point. This script was never intended to be permanent, and it attempted to prevent downloading each time, but thats not as important as repeatable and reliable.

Changed method to remove directory if present and redownload again.